keyrings/alt/file_base.py

```diff
@@ -71,35 +108,44 @@ def get_password(self, service, username):
         try:
             password_base64 = config.get(service, username).encode()
             # decode with base64
-            password_encrypted = base64.decodestring(password_base64)
-            # decrypted the password
-            password = self.decrypt(password_encrypted).decode('utf-8')
+            password_encrypted = decodebytes(password_base64)
+            # decrypt the password with associated data
+            try:
+                password = self.decrypt(password_encrypted, assoc).decode('utf-8')
+            except ValueError:
+                # decrypt the password without associated data
+                password = self.decrypt(password_encrypted).decode('utf-8')
         except (configparser.NoOptionError, configparser.NoSectionError):
             password = None
         return password
 
     def set_password(self, service, username, password):
         """Write the password in the file.
         """
-        service = escape_for_ini(service)
-        username = escape_for_ini(username)
-
+        assoc = (escape_for_ini(service) + '\0' +
```

Comment:
I can consolidate this in a method _generate_assoc, but would like to keep the null byte, since I'm using these keyrings in other projects already... Blame it on my age :wink: